### DISCUSSION TITLE
How to set  colors to background color of TextColumn and Select controls?

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#6910] [Help] [answered]
In laravel 9 app with  filament 2.17.49 I have demands table with priority_level field(in migrations defined as enum with 6 values)
and I need depending on value in priority_level field to show different background color both in table :

```
     TextColumn::make('priority_level')->sortable()->searchable()
     ...

```

and in  the form :
```
     Select::make('priority_level')
         ->preload()
         ->options(Ticket::getPriorityLevelOptions())
         ->required(),

```
priority_level Keys=> colors are written in config file. I mean different colors, not standard danger, primary, success, warning...

How can I apply these colors to background color of TextColumn and Select controls?

**Comment by RX-77IT** [comment, marked as answer]
Like [pboivin wrote in your previous thread](https://github.com/filamentphp/filament/discussions/6902#discussioncomment-6344114), have you tried `extraAttributes()`?

On the TextColumn it should be easy ([already did it once in this past thread](https://github.com/filamentphp/filament/discussions/6487)):

```php
use Filament\Tables;
use Illuminate\Database\Eloquent\Model;

public static function table(Table $table): Table
{
    return $table
        ->columns([
            Tables\Columns\TextColumn::make('priority_level')
                // ...
                // there is also: ->extraCellAttributes(function (?Model $record) {
                ->extraAttributes(function (?Model $record) {
                    $bgColor = '#5cbbc180'; // your logic to fetch the right color using the Enum here
                    // dd($record) // you can use $record to access the current model of the row
                    return ['style' => "background-color: {$bgColor};border-radius: 10px;"];
                }),
        ]);
}
```
On the Select Field we have a problem: `extraAttributes()` only modifies the `<select>` tag attributes, not the `<option>`s attributes.

So... it's not pretty with the Select open, but you can start trying with this. Maybe there is some other way I haven't considered?

```php
use Filament\Forms;

public static function form(Form $form): Form
{
return $form
    ->schema([
        Forms\Components\Select::make('priority_level')
            // ...
            ->reactive() // to reload the Select field (and the page) after changing the $state
            ->extraAttributes(function ($state) {
                $bgColor = '#5cbbc180'; // your logic to fetch the right color using the Enum here
                // dd($state) // you can use $state to access the current state of the field
                return ['style' => "background-color: {$bgColor}"];
            })
    ]);
}
```
